src/custom/components/Tokens/TokensTable.tsx

```diff
@@ -25,11 +37,36 @@ export enum TableType {
   FAVOURITE = 'FAVOURITE',
 }
 
+const _filterCb = (token: Token, query?: string) => {
+  if (!query) return false
+
+  const cleanQuery = query.toLowerCase()
+  const address = token.address.toLowerCase()
+  const symbol = token.symbol?.toLowerCase()
+  const name = token.name?.toLowerCase()
+
+  return address.match(cleanQuery) || symbol?.match(cleanQuery) || name?.match(cleanQuery)
+}
+
 export default function TokenTable({
-  tokensData,
+  tokensData: rawTokensData,
   maxItems = MAX_ITEMS,
   tableType = TableType.OVERVIEW,
 }: TokenTableParams) {
+  // search - takes precedence re:filtering
+  const [query, setQuery] = useState<string>()
+  const debouncedQuery = useDebounce(query, 300)
+
+  const handleChange = useCallback((event) => {
+    const { value } = event.target
+    setQuery(value)
```

Comment:
Here I would pass either `event.target.value` or do a parameter destructuring `useCallback(({ target: { value } }) => {` but this also works so all good.